modules/navigation_mesh_generator/godot_navigation_mesh_generator.cpp

```diff
@@ -813,19 +507,108 @@ void NavigationMeshGenerator::bake(Ref<NavigationMesh> p_navigation_mesh, Node *
 	if (ep) {
 		memdelete(ep);
 	}
-#endif
+#endif // TOOLS_ENABLED
+#endif // _3D_DISABLED
 }
 
-void NavigationMeshGenerator::clear(Ref<NavigationMesh> p_navigation_mesh) {
+void GodotNavigationMeshGenerator::clear(Ref<NavigationMesh> p_navigation_mesh) {
 	if (p_navigation_mesh.is_valid()) {
 		p_navigation_mesh->clear_polygons();
 		p_navigation_mesh->set_vertices(Vector<Vector3>());
 	}
 }
 
-void NavigationMeshGenerator::_bind_methods() {
-	ClassDB::bind_method(D_METHOD("bake", "navigation_mesh", "root_node"), &NavigationMeshGenerator::bake);
-	ClassDB::bind_method(D_METHOD("clear", "navigation_mesh"), &NavigationMeshGenerator::clear);
+#ifndef _3D_DISABLED
+void GodotNavigationMeshGenerator::register_geometry_parser_3d(Ref<NavigationGeometryParser3D> p_geometry_parser) {
+	if (geometry_3d_parsers.find(p_geometry_parser) < 0) {
+		// add front so user custom parsers that extend existing node types are called before build-in parsers
+		geometry_3d_parsers.insert(0, p_geometry_parser);
+	}
 }
 
+void GodotNavigationMeshGenerator::unregister_geometry_parser_3d(Ref<NavigationGeometryParser3D> p_geometry_parser) {
+	geometry_3d_parsers.erase(p_geometry_parser);
+}
+#endif // _3D_DISABLED
+
+void GodotNavigationMeshGenerator::cleanup() {
+#ifndef _3D_DISABLED
+	geometry_3d_parsers.clear();
+#endif // _3D_DISABLED
+}
+
+Ref<NavigationMeshSourceGeometryData> GodotNavigationMeshGenerator::parse_source_geometry_data(Ref<NavigationMesh> p_navigation_mesh, Node *p_root_node) {
```

Comment:
We seem to have two duplicate implementations of parsing the `SceneTree` to gather mesh data, this method which uses `NavigationMeshSourceGeometryData` and `bake`, which doesn't.

We should simplify to a single implementation here, instead of having two approaches.

Reply:
That was a fail from going back a few versions. Fixed now to only have one static function for each specific task.